enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,677 @@
+- the default `edge` machine pool instance type must be common in the current
+  availability zone offerings. The default for the `edge` pool will be `c5d.2xlarge`.
+- the default `edge` machine pool EBS type must be common in the current
+  availability zone offerings. The default for the `edge` pool will be `gp2`.
+
+#### Examples configuration for user-provided network
+
+Those are two examples of a user-provided network when installing a cluster
+in existing VPC:
+
+1) `install-config.yaml` with Local Zone subnets, and empty `edge` machine pool
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - subnetID-us-east-1a
+    - subnetID-us-east-1b
+    - subnetID-us-east-1c
+    - subnetID-us-east-1-nyc-1a
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+2) `install-config.yaml` with Local Zone subnets, and custom instance type
+
+> Note: the instance type `c5d.2xlarge` common between locations
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+compute:
+- name: edge
+  platform:
+    aws:
+      type: c5d.2xlarge
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - subnetID-us-east-1a
+    - subnetID-us-east-1b
+    - subnetID-us-east-1c
+    - subnetID-us-east-1-nyc-1a
+    - subnetID-us-east-1-mia-1a
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+3) `install-config.yaml` with Local Zone subnets, and custom volume type
+
+> Note: currently only `us-west-2` supports `gp3`
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+compute:
+- name: edge
+  platform:
+    aws:
+      rootVolume:
+        type: gp3
+platform:
+  aws:
+    region: us-west-2
+    subnets:
+    - subnetID-us-west-2a
+    - subnetID-us-west-2b
+    - subnetID-us-west-2c
+    - subnetID-us-west-2-lax-1a
+    - subnetID-us-west-2-lax-1b
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+The `install-config.yaml` exemplified above will generate one MachineSet
+by subnet ID on install directory `<install-dir>/openshift/`:
+
+Example  `openshift/99_openshift-cluster-api_worker-machineset-3.yaml`:
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: ${CLUSTER_ID}
+  name: ${CLUSTER_ID}-edge-us-east-1-nyc-1a  (1)
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: ${CLUSTER_ID}
+      machine.openshift.io/cluster-api-machineset: ${CLUSTER_ID}-edge-us-east-1-nyc-1a (1)
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: ${CLUSTER_ID}
+        machine.openshift.io/cluster-api-machine-role: edge  (2)
+        machine.openshift.io/cluster-api-machine-type: edge  (2)
```

Comment:
I wonder how many places have semantics about valid values for these labels 🤔 Currently it is Infra, Worker, Master and that should be it?

Reply:
Cool that makes sense, might still be nice to have the edge role on the machines though, we do the same for infra so I don't see why we shouldn't be consistent with that here

Reply:
My comment was more along the lines of, we should really try this out to see if anything starts complaining

Reply:
@JoelSpeed [this is the Installer PR](https://github.com/openshift/installer/pull/6371#issuecomment-1260409687), the machines were created with the edge role though  - your suggestions above.
(also I appreciate if you would like to take a look/review/feedback there =) )

Reply:
Looks like it should work with the edge role